owncloudData/src/main/java/com/owncloud/android/data/capabilities/datasources/implementation/OCLocalCapabilitiesDataSource.kt

```diff
@@ -19,29 +19,95 @@
 
 package com.owncloud.android.data.capabilities.datasources.implementation
 
+import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Transformations
 import com.owncloud.android.data.capabilities.datasources.LocalCapabilitiesDataSource
-import com.owncloud.android.data.capabilities.datasources.mapper.OCCapabilityMapper
 import com.owncloud.android.data.capabilities.db.OCCapabilityDao
+import com.owncloud.android.data.capabilities.db.OCCapabilityEntity
+import com.owncloud.android.domain.capabilities.model.CapabilityBooleanType
 import com.owncloud.android.domain.capabilities.model.OCCapability
 
 class OCLocalCapabilitiesDataSource(
     private val ocCapabilityDao: OCCapabilityDao,
-    private val ocCapabilityMapper: OCCapabilityMapper
 ) : LocalCapabilitiesDataSource {
 
     override fun getCapabilitiesForAccountAsLiveData(accountName: String): LiveData<OCCapability?> =
         Transformations.map(ocCapabilityDao.getCapabilitiesForAccountAsLiveData(accountName)) { ocCapabilityEntity ->
-            ocCapabilityMapper.toModel(ocCapabilityEntity)
+            ocCapabilityEntity.toModel()
         }
 
     override fun getCapabilityForAccount(accountName: String): OCCapability? =
-        ocCapabilityMapper.toModel(ocCapabilityDao.getCapabilitiesForAccount(accountName))
+        ocCapabilityDao.getCapabilitiesForAccount(accountName)?.toModel()
 
     override fun insert(ocCapabilities: List<OCCapability>) {
         ocCapabilityDao.replace(
-            ocCapabilities.map { ocCapability -> ocCapabilityMapper.toEntity(ocCapability)!! }
+            ocCapabilities.map { ocCapability -> ocCapability.toEntity() }
         )
     }
+
+    companion object {
+        @VisibleForTesting
+        fun OCCapabilityEntity.toModel(): OCCapability =
+            OCCapability(
+                id = id,
+                accountName = accountName,
+                versionMayor = versionMayor,
+                versionMinor = versionMinor,
+                versionMicro = versionMicro,
+                versionString = versionString,
+                versionEdition = versionEdition,
+                corePollInterval = corePollInterval,
+                davChunkingVersion = davChunkingVersion,
```

Comment:
Maybe you could remove the named params here since are using the same name

Reply:
Well, I like the named params even if we use the same name. And there is no counterpart of using them. So, I would keep them 👍